core/tests/protractor_desktop/topicsAndSkillsDashboard.js

```diff
@@ -85,62 +89,95 @@ describe('Topics and skills dashboard functionality', function() {
       Constants.SKILL_STATUS_UNASSIGNED);
     await topicsAndSkillsDashboardPage.expectNumberOfSkillsToBe(1);
     await topicsAndSkillsDashboardPage.resetTopicFilters();
-    await topicsAndSkillsDashboardPage.deleteSkillWithIndex(0);
+    await topicsAndSkillsDashboardPage.deleteSkillWithName(skillName);
     await topicsAndSkillsDashboardPage.get();
     await topicsAndSkillsDashboardPage.navigateToSkillsTab();
+    await topicsAndSkillsDashboardPage.searchSkillByName(skillName);
     await topicsAndSkillsDashboardPage.expectNumberOfSkillsToBe(0);
 
     await topicsAndSkillsDashboardPage.get();
+    await topicsAndSkillsDashboardPage.filterTopicsByKeyword(topicName);
     await topicsAndSkillsDashboardPage.expectNumberOfTopicsToBe(1);
-    await topicsAndSkillsDashboardPage.deleteTopicWithIndex(0);
-    await topicsAndSkillsDashboardPage.expectNumberOfTopicsToBe(0);
+    await topicsAndSkillsDashboardPage.deleteTopicWithName(topicName);
+    let topicsTableIsPresent = (
+      await topicsAndSkillsDashboardPage.isTopicTablePresent());
+    if (topicsTableIsPresent) {
+      await topicsAndSkillsDashboardPage.filterTopicsByKeyword(topicName);
+      await topicsAndSkillsDashboardPage.expectNumberOfTopicsToBe(0);
+    }
   });
 
   it('should filter the topics', async function() {
-    await topicsAndSkillsDashboardPage.expectNumberOfTopicsToBe(0);
+    let topicNameAlpha = 'Alpha TASD';
+    let topicNameBeta = 'Beta TASD';
+    let topicsTableIsPresent = (
+      await topicsAndSkillsDashboardPage.isTopicTablePresent());
+    if (topicsTableIsPresent) {
+      await topicsAndSkillsDashboardPage.filterTopicsByKeyword(topicNameAlpha);
+      await topicsAndSkillsDashboardPage.expectNumberOfTopicsToBe(0);
+      await topicsAndSkillsDashboardPage.resetTopicFilters();
+      await topicsAndSkillsDashboardPage.filterTopicsByKeyword(topicNameBeta);
+      await topicsAndSkillsDashboardPage.expectNumberOfTopicsToBe(0);
+    }
     await topicsAndSkillsDashboardPage.createTopic(
-      'Alpha TASD', 'alpha-tasd', 'Alpha description', true);
+      topicNameAlpha, 'alpha-tasd', 'Alpha description', true);
     await topicsAndSkillsDashboardPage.get();
     await topicsAndSkillsDashboardPage.createTopic(
-      'Beta TASD', 'beta-tasd', 'Beta description', true);
+      topicNameBeta, 'beta-tasd', 'Beta description', true);
 
     await topicsAndSkillsDashboardPage.get();
+    await topicsAndSkillsDashboardPage.filterTopicsByKeyword(topicNameAlpha);
+    await topicsAndSkillsDashboardPage.filterTopicsByKeyword(topicNameBeta);
     await topicsAndSkillsDashboardPage.expectNumberOfTopicsToBe(2);
+    await topicsAndSkillsDashboardPage.resetTopicFilters();
+
     await topicsAndSkillsDashboardPage.filterTopicsByKeyword('alp');
     await topicsAndSkillsDashboardPage.expectNumberOfTopicsToBe(1);
     await topicsAndSkillsDashboardPage.resetTopicFilters();
-    await topicsAndSkillsDashboardPage.expectNumberOfTopicsToBe(2);
```

Comment:
Do we want to check that resetting the filters works? Maybe we could count how many are visible at the beginning and use that instead of 2